ociclient/cache/cache.go

```diff
@@ -230,9 +232,11 @@ func (lc *layeredCache) getFromOverlay(dgst string, desc ocispecv1.Descriptor) (
 		return nil, nil, ErrNotFound
 	}
 
-	// do simple blob verification
-	// todo: use digest to check if blob is valid
-	if info.Size() != desc.Size {
+	verified, err := verifyBlob(lc.overlayFs.FileSystem, info, dgst, desc)
+	if err != nil {
+		return nil, nil, fmt.Errorf("unable to verify blob: %w", err)
```

Comment:
Could there be an endless loop if blob in cache is damaged?

Reply:
same as above